src/Microsoft.Data.Analysis/DataFrame.cs

```diff
@@ -429,6 +429,10 @@ public void Append(IEnumerable<object> row = null)
                 {
                     DataFrameColumn column = columnEnumerator.Current;
                     object value = rowEnumerator.Current;
+                    if (value is string stringValue && string.IsNullOrEmpty(stringValue) && column.DataType != typeof(string))
```

Comment:
Should probably add a comment here on why we are doing this - and why not for strings.